### PR TITLE
WEB-411: Additional helpers and documentation for layouts, typography, lists, and responsiveness

### PR DESCRIPTION
**Pull request by atla5**
Jira Ticket: [WEB-411](https://bulibrary.atlassian.net/browse/WEB-411)

### Background
- there are numerous helper classes (both functional and object-oriented) that are used throughout our sites to quickly and cheaply add generalized (and often schema-independent) styling
- to date, these are largely undocumented outside of the `assets/css/*.css` files themselves, and therefore tend to go unnoticed
- as we go forward, we'll want greater visibility into some of these helpers as other content authors with less familiarity with the site want to add or augment new pages/parts of pages
- this is particularly relevant for the creation of more advanced styling (reminiscent of other peer library sites) as 'composites'

### Description of Changes
- [x] extract various helpers from `common.css` to more semantic filenames
- [x] add documentation particularly for layouts and typography, especially as it relates to responsiveness
- [x] create new 'composites' section to demonstrate more advanced usage of the helpers

### Screenshots

screenshot of storybook with new pages

### CODE REVIEW

File: sites/libguides/libguides.css [hunk 1]
```diff
@@ -37,6 +37,7 @@ body { padding-left: 0px; padding-right: 0px; }
 /* normalize width alignment of top parts to match the header and footer breakpoints */
 @media only screen and (min-width: 1500px){
   #s-lg-guide-header #s-lg-guide-header-info > h1,
+    #s-lg-guide-header #s-lib-public-header-title > h1,
     #s-lg-guide-header #s-lg-guide-header-info > div#s-lg-guide-desc-container {
```
Comment (atla5) [inline sites/libguides/libguides.css:41]: note: this is to address a libguides bug with search and profile pages: 

image

File: src/_helpers/lib_info_helper.js [hunk 2]
```diff
@@ -138,7 +138,7 @@ function getSiteCodeFromUrl(url, debug, defSiteCode){
   if(url.includes("askalibrarian")){ site_code = "help"; }
   else if(url.includes("/disc/") || url.includes("/dioa")){ site_code ="services"; }
   else if(url.includes("open.bu") || url.includes("archives")){ site_code = "collections"; }
-  else if(url.includes("guides")){ site_code = "guides" }
+  else if(url.includes("guides") || url.includes("library.bu.edu")){ site_code = "guides" }
```
Comment (atla5) [inline src/_helpers/lib_info_helper.js:141]: another bugfix for guides pages not getting the guides footer

**Comment by mwardlib** [comment]
reviewed, looks good.

**Review by mwardlib — APPROVED** [review]
approved

**Pull request merged**